tests/it/io/ipc/read/mmap.rs

```diff
@@ -0,0 +1,40 @@
+use arrow2::error::Result;
+use arrow2::io::ipc::mmap::map_chunk;
+
+use super::super::common::read_gzip_json;
+
+#[derive(Clone)]
```

Comment:
When we mmap we never get an owned `Vec`, only a `&[u8]` or a `&mut [u8]`. Do we create this `Vec` temporarily and simply never allow it to drop?

We might also store the `memmap2::Mmap` struct. This implements `AsRef<[u8]>` and will ensure the memory mapped region remains mapped until it is dropped.

Reply:
yeah, but `&[u8]` is technically `&'a[u8]` and our arrays are owned (they do not carry any lifetime). So, we need to control lifetime somewhere else, which is why we need an owned version of the `AsRef<[u8]>`. Moving `memmap2::Mmap` to `Arc<_>` is the most natural choice.

Reply:
> Moving memmap2::Mmap to Arc<_> is the most natural choice.

Agreed.